packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+        });
+        expect(pages.config.isSupported()).toBeTruthy();
+      });
+    });
+  });
+
+  describe('Testing pages.backStack namespace', () => {
+    describe('Testing pages.backStack._initialize function', () => {
+      it('pages.backStack._initialize should successfully register backButtonPress handler', () => {
+        pages.backStack._initialize();
+        const message = utils.findMessageByFunc('backButtonPress');
+        expect(message).toBeNull();
+      });
+    });
+
+    describe('Testing pages.backStack.navigateBack function', () => {
+      it('pages.backStack.navigateBack should not allow calls before initialization', async () => {
+        await expect(pages.backStack.navigateBack()).rejects.toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`pages.backStack.navigateBack should register the navigateBack action when initialized with ${context} context`, () => {
+          utils.initializeWithContext(context);
+          pages.backStack.navigateBack();
+          const navigateBackMessage = utils.findMessageByFunc('navigateBack');
+          expect(navigateBackMessage).not.toBeNull();
+        });
+      });
+    });
+
+    describe('Testing pages.backStack.registerBackButtonHandler function', () => {
+      Object.values(FrameContexts).forEach(context => {
+        it('pages.backStack.registerBackButtonHandler should successfully register a back button handler and not call navigateBack if it returns true', async () => {
+          await utils.initializeWithContext(context);
+
+          let handlerInvoked = false;
+          pages.backStack.registerBackButtonHandler(() => {
+            handlerInvoked = true;
+            return true;
+          });
+
+          utils.sendMessage('backButtonPress');
+
+          let navigateBackMessage = utils.findMessageByFunc('navigateBack');
+          expect(navigateBackMessage).toBeNull();
+          expect(handlerInvoked).toBe(true);
+        });
+
+        it('pages.backStack.registerBackButtonHandler should successfully register a back button handler and call navigateBack if it returns false', async () => {
+          await utils.initializeWithContext(context);
+
+          let handlerInvoked = false;
+          pages.backStack.registerBackButtonHandler(() => {
+            handlerInvoked = true;
+            return false;
+          });
+
+          utils.sendMessage('backButtonPress');
+
+          let navigateBackMessage = utils.findMessageByFunc('navigateBack');
+          expect(navigateBackMessage).not.toBeNull();
+          expect(handlerInvoked).toBe(true);
+        });
+      });
+    });
+
+    describe('Testing pages.backStack.isSupported function', () => {
+      it('pages.backStack.isSupported should return false if the runtime says pages.backStack is not supported', () => {
+        utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
```

Comment:
Can you please add a test case that verifies that `pages.backStack.isSupported` returns false when we set the runtime config to `supports: {}`?

Reply:
Added in the latest commit